Fix inconsistent tabbing in docs navbar

dcastil opened the pull request:
While working on #943 I noticed some inconsistencies in the docs navbar.

* Multiple outline styles for different buttons
* SkipLink doesn't follow color scheme which makes it very obtrusive on dark mode

I opened a PR right away because the changes could be made quickly. If those things are on purpose, feel free to close this PR. 😊

| Before | After |
| --- | --- |
| ![Chrome light before](https://user-images.githubusercontent.com/31006608/82604194-355e1c00-9bb4-11ea-82a3-b7a7e2082f74.gif) | ![Chrome light after](https://user-images.githubusercontent.com/31006608/82604419-95ed5900-9bb4-11ea-9f24-68eecaa684d4.gif) |
| ![Safari light before](https://user-images.githubusercontent.com/31006608/82604218-4018b100-9bb4-11ea-8602-b571e3f619db.gif) | ![Safari light after](https://user-images.githubusercontent.com/31006608/82604429-9a197680-9bb4-11ea-87a5-08fc030527d9.gif) |
| ![Chrome dark before](https://user-images.githubusercontent.com/31006608/82604306-68081480-9bb4-11ea-9d76-52dbc143da0f.gif) | ![Chrome dark after](https://user-images.githubusercontent.com/31006608/82604646-e5338980-9bb4-11ea-9bfe-87cf83527530.gif) |
| ![Safari dark before](https://user-images.githubusercontent.com/31006608/82604316-6c343200-9bb4-11ea-804a-1135c2e8e00b.gif) | ![Safari dark after](https://user-images.githubusercontent.com/31006608/82604660-e95fa700-9bb4-11ea-93d8-e719945343f7.gif) |

jxnblk commented:
Thanks for this! These were intentional, but could you elaborate on what you're trying to improve upon here or what the issue was?

dcastil commented:
I tried to solve two issues I had with the tabbing experience:

### Outlines

The outline around a focused element looks different dependent on the focused element.The hamburger menu and the color mode switch have a thicker black outline with sharp edges while the rest of the elements has the browser-native blue-ish outline with rounded corners.

I find a consistent outline more aesthetically pleasing and possibly less confusing (that part is negligible in the light theme, but might become more important with other themes).

So I removed the custom outline from both hamburger menu and color mode switch.

### SkipLink

The SkipLink is black text on white background, regardless of color mode.

I usually use dark mode on websites when available and get used to the dark look very quickly. I find it very unpleasing when there is sudden swap in text-color and bg-color. It always takes a little while so my eyes get used to read text that way (the problem is much bigger when reading a light section in a dark-themed website than the other way around). A good negative example is the dark mode the [TypeScript website](https://www.staging-typescript.org/docs/handbook/typescript-in-5-minutes.html) (visit it with dark mode enabled in your OS) implemented.

<img width="902" alt="Screenshot 2020-05-22 at 21 57 17" src="https://user-images.githubusercontent.com/31006608/82705078-373ce380-9c77-11ea-98c4-ec9a76a87cd2.png">

I fixed it by using the current `text` and `background` colors.

dcastil commented:
Another option would be to add a custom outline to all elements instead of using the default one. That would make the look more consistent across browsers and fix the low contrast in the Safari outline in dark mode.

But then it's also easy to forget to implement that on any new element which would make it less consistent.

jxnblk commented:
Thanks for the context! For the skip nav link, part of the rationale in keeping that black on white is to be more visible and appear as expected for users who might have colors inverted or custom styles -- i.e. I think it's good for that one to not match custom styles, since they might pose issues.

For the other focus states, I think making them more consistent makes sense, so maybe those all can rely on browser defaults. What do you think?

dcastil commented:
Good to know, I agree with that! I removed the changes from SkipLink.
